interface/core.xhr.json

```diff
@@ -0,0 +1,41 @@
+{
+  "name": "core.xhr",
+  "api": {
+    // @param {String} method - GET/POST/DELETE
+    // @param {String} url
+    // @param {Boolean} async
+    // @param {String} user
+    // @param {String} password
+    "open": { "type": "method", "value": [ "string", "string", "boolean", "string", "string" ] },
+    // Specify a single element in the argument object,
+    // depending on the type of data to send (e.g. in POST)
+    "send": { "type": "method", "value": [ { "string": "string", "buffer": "buffer" } ] },
+    "abort": { "type": "method", "value": [] },
+    // @param {String} header
+    // @return {String} value
+    "getResponseHeader": { "type": "method", "value": [ "string" ], "ret": "string" },
+    // @param {String} all headers in a String
+    "getAllResponseHeaders": { "type": "method", "value": [], "ret": "string" },
+    // @param {String} header
+    // @param {String} value
+    "setRequestHeader": { "type": "method", "value": [ "string", "string" ] },
+    // @param {String} mimetype
+    "overrideMimeType": { "type": "method", "value": [ "string" ] },
+    "getReadyState": { "type": "method", "value": [], "ret": "number" },
+    "getResponse": { "type": "method", "value": [], "ret": { "string": "string", "buffer": "buffer" } },
+    "getResponseText": { "type": "method", "value": [], "ret": "string" },
+    "getResponseType": { "type": "method", "value": [], "ret": "string" },
```

Comment:
I think you also need a `setResponseType` method, because the purpose of `responseType` is to allow the user to [choose the return type](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest#responseType).  Otherwise, `responseType` will always be the default (i.e. string).